src/pretix/control/forms/waitinglist.py

```diff
@@ -54,3 +60,97 @@ class Meta:
         field_classes = {
             'subevent': SafeModelChoiceField,
         }
+
+
+class WaitingListEntryEditForm(I18nModelForm):
+    itemvar = ChoiceField()
+
+    def __init__(self, *args, **kwargs):
+        self.instance = kwargs.get('instance', None)
+        initial = kwargs.get('initial', {})
+        if self.instance and self.instance.pk and 'itemvar' not in initial:
+            if self.instance.variation is not None:
+                initial['itemvar'] = f'{self.instance.item.pk}-{self.instance.variation.pk}'
+            else:
+                initial['itemvar'] = self.instance.item.pk
+        kwargs['initial'] = initial
+
+        super().__init__(*args, **kwargs)
+
+        # Prevent the item field cleaning from complaining that it isn't populated
+        # the value for item is derived and populated during form.clean()
+        self.fields['item'].required = False
+
+        self.fields['name_parts'] = NamePartsFormField(
+            max_length=255,
+            required=True,
+            scheme=self.event.organizer.settings.name_scheme,
+            titles=self.event.organizer.settings.name_scheme_titles,
+            label=_('Name'),
+        )
+
+        self.fields['name_parts'].required = self.event.settings.waiting_list_names_required
+        self.fields['name_parts'].one_required = self.event.settings.waiting_list_names_required
+        self.fields['phone'].required = self.event.settings.waiting_list_phones_required
+
+        choices = []
+
+        items = self.event.items.prefetch_related('variations')
+        for item in items:
+            if len(item.variations.all()) > 0:
+                for v in item.variations.all():
+                    choices.append((
+                        '{}-{}'.format(item.pk, v.pk),
+                        '{} – {}'.format(item, v.value) if item.active else mark_safe(
+                            f'<strike class="text-muted">{escape(item)} – {escape(v.value)}</strike>')
+                    ))
+            else:
+                choices.append(('{}'.format(item.pk), str(item) if item.active else mark_safe(
+                    f'<strike class="text-muted">{escape(item)}</strike>')))
+
+        self.fields['itemvar'].label = _("Item and Variation")
```

Comment:
In most of the other cases it is simply called „Product“, so I'd reuse that here.
```suggestion
        self.fields['itemvar'].label = _("Product")
```